core/src/Revolution/Processors/Resource/Trash/GetList.php

```diff
@@ -119,8 +119,13 @@ public function prepareRow(xPDOObject $object)
         $parent = $objectArray['parent'];
 
         while ($parent != 0) {
-            $parents[] = $this->modx->getObject(modResource::class, $parent);
-            $parent = end($parents)->get('parent');
+            $o = $this->modx->getObject('modResource', $parent);
+            if($o) {
+                $parents[] = $o;
+            } else {
+                break;
+            }
+            if($parents) $parent = end($parents)->get('parent');
```

Comment:
Seems like this code could be simplified:

```
            $o = $this->modx->getObject(modResource::class, $parent);
            if ($o) {
                $parents[] = $o;
                $parent = $o->get('parent');
            } 
            else {
                break;
            }
```

Reply:
Note that in MODX3 the `modResource::class` syntax should be used rather than `'modResource'`.

Reply:
It would be nice if the name of the variable `$o` were to be changed to something more descriptive.